### ISSUE TITLE
Incomplete documentation for reexported data type from internal library

### ISSUE THREAD

**Issue by vrom911** [haskell/haddock#876]
I've noticed that haddock for the types that are exported from the internal library is not shown properly.

## Examples

Here is the simplest example of this behaviour: 
* https://github.com/vrom911/haddock-example

But it was discovered in this library first, in case you need to check: 
* https://github.com/vrom911/typerep-map/

## Steps to reproduce:

* Clone the repository
* inside the repository folder run `cabal new-haddock` command
* Check the docs and see that what is there:

<img width="778" alt="wrong" src="https://user-images.githubusercontent.com/8126674/42432366-cd40c2a4-837c-11e8-81ea-4caed4fd9d66.png">

While what I want to get is:

<img width="769" alt="right" src="https://user-images.githubusercontent.com/8126674/42432365-cd115172-837c-11e8-85f0-4689700fca7c.png">

## System information

In this project, I'm using `ghc-8.4.3`. 
My machine is MacOS High Sierra v. 10.13.5.

```
$ cabal --version
cabal-install version 2.2.0.0
compiled using version 2.2.0.0 of the Cabal library
```

**Comment by harpocrates** [comment]
I believe this is already fixed, as I can replicate this in the `ghc-8.4` branch, but not the `ghc-head` branch, or in the Haddock bundled with the GHC 8.6 alpha release.

---

Here is a one file test case (using the `Coercion` data type in `base`):

```
module ExistentialReexported (Coercion(..)) where

import Data.Type.Coercion
```

On the Haddock bundled with `ghc-8.4`:

<img width="608" alt="screen shot 2018-07-09 at 7 17 28 am" src="https://user-images.githubusercontent.com/10766081/42455844-2cd9e5f4-8348-11e8-96f7-895edbb6ad0e.png">

On the Haddock bundled with `ghc-8.6.0.20180627` (the alpha release):

<img width="598" alt="screen shot 2018-07-09 at 7 17 02 am" src="https://user-images.githubusercontent.com/10766081/42455875-464a2224-8348-11e8-9e16-6f0935cc6287.png">

**Comment by harpocrates** [comment]
Closing this, since `haddock-ghc-8.6.1` has been release.
